runtime/oti/ObjectMonitor.hpp

```diff
@@ -284,6 +284,11 @@ class VM_ObjectMonitor
 #endif /* JAVA_SPEC_VERSION >= 16 */
 		) {
 			locked = inlineFastInitAndEnterMonitor(currentThread, J9OBJECT_MONITOR_EA(currentThread, object));
+#if JAVA_SPEC_VERSION >= 19
```

Comment:
This would be better put in `inlineFastInitAndEnterMonitor` which requires changes to the cpp as well. It's much eaeier to understand if the inc/dec is in the places where the monitor slot is written.

Reply:
`inlineFastInitAndEnterMonitor` gets invoked from `objectMonitorEnterBlocking` (two locations). There are code paths in `objectMonitorEnterBlocking` where `inlineFastInitAndEnterMonitor` does not get invoked during exit. We may miss to increment the counter if one of these code paths are taken.

Reply:
I still think the most obvious implementation is to tie the counter to modification of the lockword as closely as possible.

Reply:
Updated. See https://github.com/eclipse-openj9/openj9/pull/15487#issuecomment-1178208623.